### ISSUE TITLE
Device doesn't reconnect after router disconnects

### ISSUE THREAD

**Issue by radredgreen** [paullj1/TreatLife-HomeKit#4]
Hi Paul,

I'm working on my branch with the themostat that I've added and have noticed strange behavior that I'm wondering if you're experiencing too.

**Describe the bug**
When the wifi router disconnects the ethernet connection (for example during a wifi reboot) the esp8266 disconnects from the WIFI and never reconnects until reboot.

**To Reproduce**
1. Connect, pair, and get working as normal
2. Disconnect Wifi router by, for example, rebooting the wifi router
3. Try to connect to the device over home kit

**Expected behavior**
Expecting the esp to reconnect to wifi after disconnect

**Build Settings**
Working on the Saswell thermostat branch on my repo, although I expect functionality to be consistent across the two repos.

**Additional context**
I wonder if wifimanager has some boilerplate code for reconnecting after a disconnect.

Thanks!
Radredgreen

**Comment by paullj1** [comment]
Hi @radredgreen, I’ve actually not experienced this, though I have seen interesting behavior (for which I have an open issue on the HomeKit repo) when adding someone to the home.  I actually just had my router reboot the other day. It could be that I didn’t try to immediately connect to my devices running this firmware, but the next time I tried it worked. I believe there is logic within the WiFi Manager code to handle reconnects. Did it ever come back? It may take a few minutes.

**Comment by radredgreen** [comment]
I'm not sure exactly what was causing the issue.  It would reconnect to the wifi router and acquire a DHCP lease.  The MDNS announcements were not made, but the HomeKit port (udp 6555?) was open as reported by nmap.

I added a reboot upon detection of a disconnect which is  a stop gap fix and has worked through several router restarts.  

Closing this issue since this fix seems to work and I don't know enough to further debug the MDNS issue.
